packages/foam-vscode/src/extension.ts

```diff
@@ -33,6 +36,57 @@ export async function activate(context: ExtensionContext) {
 
     // Prepare Foam
     const excludes = getIgnoredFilesSetting().map(g => g.toString());
+
+    Logger.info('[wtw] Excluded patterns from settings: ' + excludes);
+
+    // Read all .gitignore files and add patterns to excludePatterns
+    const gitignoreFiles = await workspace.findFiles('**/.gitignore');
+
+    gitignoreFiles.forEach(gitignoreUri => {
+      try {
+        workspace.fs.stat(gitignoreUri); // Check if the file exists
+        const gitignoreContent = await workspace.fs.readFile(gitignoreUri);
+
+        // TODO maybe better to use a specific gitignore parser lib.
+        let ignore_rules = Buffer.from(gitignoreContent)
+          .toString('utf-8')
+          .split('\n')
+          .map(line => line.trim())
+          .filter(line => !isEmpty(line))
+          .filter(line => !line.startsWith('#'))
+          .forEach(line => excludes.push(line));
+
+        excludes.push(...ignore_rules);
+        Logger.info(
+          `Excluded patterns from ${gitignoreUri.path}: ${ignore_rules}`
+        );
+      } catch (error) {
+        Logger.error(`Error reading .gitignore file: ${error}`);
+      }
+    });
```

Comment:
@riccardoferretti Do you think here is a reasonable place to calculate ignore rules? I tend to append them into `exclude` variable.

```log
[info - 6:03:40 PM] [wtw] Starting Foam
[info - 6:03:40 PM] [wtw] Excluded patterns from settings: **/.foam/**,**/.vscode/**/*,**/_layouts/**/*,**/_site/**/*,**/node_modules/**/*,**/.git,**/.svn,**/.hg,**/CVS,**/.DS_Store,**/Thumbs.db,**/.classpath,**/.factorypath,**/.project,**/.settings
[info - 6:03:47 PM] Excluded patterns from /tmp/test_folder/.gitignore: ignored_note.md
[info - 6:03:48 PM] Loading from directories:
[info - 6:03:48 PM] - /tmp/test_folder
[info - 6:03:48 PM]   Include: **/*
[info - 6:03:48 PM]   Exclude: ignored_note.md,**/.foam/**,**/.vscode/**/*,**/_layouts/**/*,**/_site/**/*,**/node_modules/**/*,**/.git,**/.svn,**/.hg,**/CVS,**/.DS_Store,**/Thumbs.db,**/.classpath,**/.factorypath,**/.project,**/.settings
[info - 6:03:49 PM] Workspace loaded in 1071ms
[info - 6:03:49 PM] Graph loaded in 2ms
[info - 6:03:49 PM] Tags loaded in 0ms
[info - 6:03:49 PM] [wtw] Excluded patterns from settings:
[info - 6:03:49 PM] [wtw] Excluded patterns from settings:
[info - 6:03:49 PM] Loaded 2 resources
[info - 6:03:50 PM] Excluded patterns from /tmp/test_folder/.gitignore: ignored_note.md
[info - 6:03:50 PM] Excluded patterns from /tmp/test_folder/.gitignore: ignored_note.md
```